File: AMF5/src/test/java/javaPlatform/ResourceLoaderTest.java
package javaPlatform;

import amf.apicontract.client.platform.AMFBaseUnitClient;
import amf.apicontract.client.platform.AMFConfiguration;
import amf.apicontract.client.platform.RAMLConfiguration;
import amf.core.client.common.remote.Content;
import amf.core.client.common.validation.ProfileNames;
import amf.core.client.platform.model.document.BaseUnit;
import amf.core.client.platform.resource.FileResourceLoader;
import amf.core.client.platform.resource.ResourceLoader;
import amf.core.client.platform.validation.AMFValidationReport;
import org.junit.Test;

import java.io.File;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutionException;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

public class ResourceLoaderTest {

    @Test
    public void validateRamlWithResourceLoader() throws ExecutionException, InterruptedException {
        // add custom resource loader to the configuration
        final AMFConfiguration config = RAMLConfiguration.RAML10().withResourceLoader(new CustomResourceLoader());
        final AMFBaseUnitClient client = config.baseUnitClient();

        // using a custom protocol that our CustomResourceLoader can parse
        final BaseUnit model = client.parse("CustomProtocol/src/test/resources/examples/banking-api.raml").get().baseUnit();
        assertNotNull(model);

        // Run RAML default validations on parsed unit (expects no errors).
        final AMFValidationReport report = client.validate(model).get();
        System.out.println("report.conforms() == " + report.conforms());
        assertTrue(report.conforms());
    }
}

File: AMF5/src/test/scala/scalaPlatform/ResourceLoaderTest.scala
package scalaPlatform

import amf.apicontract.client.scala.{RAMLConfiguration, WebAPIConfiguration}
import org.junit.runner.RunWith
import org.scalatest.flatspec.AsyncFlatSpec
import org.scalatest.matchers.should
import org.scalatestplus.junit.JUnitRunner

import java.util.regex.Pattern

@RunWith(classOf[JUnitRunner])
class ResourceLoaderTest extends AsyncFlatSpec with should.Matchers {

  private val CUSTOM_PATH_PATTERN = Pattern.compile("^CustomProtocol/")

  "AMF custom resource loader" should "be created and used in the configuration" in {
    val rl = new CustomResourceLoader(CUSTOM_PATH_PATTERN)
    val client = RAMLConfiguration
      .RAML10()
      .withResourceLoader(rl)
      .baseUnitClient()

    client.parse(
      "CustomProtocol/src/test/resources/examples/banking-api.raml"
    ) map (_.conforms shouldBe true)
  }
}
